package testcases.CWE523_Unprotected_Cred_Transport;

import javax.servlet.http.*;

public class ServletMain extends HttpServlet {

    private static final long serialVersionUID = 1L; /* needed since Servlets are serializable */


    public void doPost(HttpServletRequest request, HttpServletResponse response) {
        doGet(request, response);
    }

    public void doGet(HttpServletRequest request, HttpServletResponse response) {
        doGetCWE1(request, response);
        doGetCWE2(request, response);
        doGetCWE3(request, response);
        doGetCWE4(request, response);
        doGetCWE5(request, response);
        doGetCWE6(request, response);
        doGetCWE7(request, response);
        doGetCWE8(request, response);
        doGetCWE9(request, response);
    }

    private static void doGetCWE1(HttpServletRequest request, HttpServletResponse response) {
        try {
            /* BEGIN-AUTOGENERATED-SERVLET-TESTS-1 */

            /* END-AUTOGENERATED-SERVLET-TESTS-1 */
        } catch (Throwable t) {

            /* this will only happen on an IOException or something (the runTest for the test cases will catch any exceptions from the test cases).  So, we just abort
             * and write to the console (since we can't write to the response without possibly throwing another exception)
             */

            System.out.println(
                    " Caught thowable from doGetCWE1 ");

            System.out.println(" Throwable's message = " + t.getMessage());

            System.out.println("Stack trace below");

            StackTraceElement stes[] = t.getStackTrace();

            for (StackTraceElement ste : stes) {
                System.out.println("  " + ste.toString());
            }
        }
    }

    private static void doGetCWE2(HttpServletRequest request, HttpServletResponse response) {
        try {
            /* BEGIN-AUTOGENERATED-SERVLET-TESTS-2 */

            /* END-AUTOGENERATED-SERVLET-TESTS-2 */
        } catch (Throwable t) {

            /* this will only happen on an IOException or something (the runTest for the test cases will catch any exceptions from the test cases).  So, we just abort
             * and write to the console (since we can't write to the response without possibly throwing another exception)
             */

            System.out.println(
                    " Caught thowable from doGetCWE2 ");

            System.out.println(" Throwable's message = " + t.getMessage());

            System.out.println("Stack trace below");

            StackTraceElement stes[] = t.getStackTrace();

            for (StackTraceElement ste : stes) {
                System.out.println("  " + ste.toString());
            }
        }
    }

    private static void doGetCWE3(HttpServletRequest request, HttpServletResponse response) {
        try {
            /* BEGIN-AUTOGENERATED-SERVLET-TESTS-3 */

            /* END-AUTOGENERATED-SERVLET-TESTS-3 */
        } catch (Throwable t) {

            /* this will only happen on an IOException or something (the runTest for the test cases will catch any exceptions from the test cases).  So, we just abort
             * and write to the console (since we can't write to the response without possibly throwing another exception)
             */

            System.out.println(
                    " Caught thowable from doGetCWE3 ");

            System.out.println(" Throwable's message = " + t.getMessage());

            System.out.println("Stack trace below");

            StackTraceElement stes[] = t.getStackTrace();

            for (StackTraceElement ste : stes) {
                System.out.println("  " + ste.toString());
            }
        }
    }

    private static void doGetCWE4(HttpServletRequest request, HttpServletResponse response) {
        try {
            /* BEGIN-AUTOGENERATED-SERVLET-TESTS-4 */

            /* END-AUTOGENERATED-SERVLET-TESTS-4 */
        } catch (Throwable t) {

            /* this will only happen on an IOException or something (the runTest for the test cases will catch any exceptions from the test cases).  So, we just abort
             * and write to the console (since we can't write to the response without possibly throwing another exception)
             */

            System.out.println(
                    " Caught thowable from doGetCWE4 ");

            System.out.println(" Throwable's message = " + t.getMessage());

            System.out.println("Stack trace below");

            StackTraceElement stes[] = t.getStackTrace();

            for (StackTraceElement ste : stes) {
                System.out.println("  " + ste.toString());
            }
        }
    }

    private static void doGetCWE5(HttpServletRequest request, HttpServletResponse response) {
        try {
            /* BEGIN-AUTOGENERATED-SERVLET-TESTS-5 */
            (new testcases.CWE523_Unprotected_Cred_Transport.J19791()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19790()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19787()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19786()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19813()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19812()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19785()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19784()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19789()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19788()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19797()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19796()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19807()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19806()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19815()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19814()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19799()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19798()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19795()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19794()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19811()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19810()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19793()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19792()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19805()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19804()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19809()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19808()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19803()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19802()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19783()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19782()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19801()).runTest(request, response);
            (new testcases.CWE523_Unprotected_Cred_Transport.J19800()).runTest(request, response);
            /* END-AUTOGENERATED-SERVLET-TESTS-5 */
        } catch (Throwable t) {

            /* this will only happen on an IOException or something (the runTest for the test cases will catch any exceptions from the test cases).  So, we just abort
             * and write to the console (since we can't write to the response without possibly throwing another exception)
             */

            System.out.println(
                    " Caught thowable from doGetCWE5 ");

            System.out.println(" Throwable's message = " + t.getMessage());

            System.out.println("Stack trace below");

            StackTraceElement stes[] = t.getStackTrace();

            for (StackTraceElement ste : stes) {

                System.out.println("  " + ste.toString());
            }
        }
    }

    private static void doGetCWE6(HttpServletRequest request, HttpServletResponse response) {
        try {
            /* BEGIN-AUTOGENERATED-SERVLET-TESTS-6 */

            /* END-AUTOGENERATED-SERVLET-TESTS-6 */
        } catch (Throwable t) {

            /* this will only happen on an IOException or something (the runTest for the test cases will catch any exceptions from the test cases).  So, we just abort
             * and write to the console (since we can't write to the response without possibly throwing another exception)
             */

            System.out.println(
                    " Caught thowable from doGetCWE6 ");

            System.out.println(" Throwable's message = " + t.getMessage());

            System.out.println("Stack trace below");

            StackTraceElement stes[] = t.getStackTrace();

            for (StackTraceElement ste : stes) {
                System.out.println("  " + ste.toString());
            }
        }
    }

    private static void doGetCWE7(HttpServletRequest request, HttpServletResponse response) {
        try {
            /* BEGIN-AUTOGENERATED-SERVLET-TESTS-7 */

            /* END-AUTOGENERATED-SERVLET-TESTS-7 */
        } catch (Throwable t) {

            /* this will only happen on an IOException or something (the runTest for the test cases will catch any exceptions from the test cases).  So, we just abort
             * and write to the console (since we can't write to the response without possibly throwing another exception)
             */

            System.out.println(
                    " Caught thowable from doGetCWE7 ");

            System.out.println(" Throwable's message = " + t.getMessage());

            System.out.println("Stack trace below");

            StackTraceElement stes[] = t.getStackTrace();

            for (StackTraceElement ste : stes) {
                System.out.println("  " + ste.toString());
            }
        }
    }

    private static void doGetCWE8(HttpServletRequest request, HttpServletResponse response) {
        try {
            /* BEGIN-AUTOGENERATED-SERVLET-TESTS-8 */

            /* END-AUTOGENERATED-SERVLET-TESTS-8 */
        } catch (Throwable t) {

            /* this will only happen on an IOException or something (the runTest for the test cases will catch any exceptions from the test cases).  So, we just abort
             * and write to the console (since we can't write to the response without possibly throwing another exception)
             */

            System.out.println(
                    " Caught thowable from doGetCWE8 ");

            System.out.println(" Throwable's message = " + t.getMessage());

            System.out.println("Stack trace below");

            StackTraceElement stes[] = t.getStackTrace();

            for (StackTraceElement ste : stes) {
                System.out.println("  " + ste.toString());
            }
        }
    }

    private static void doGetCWE9(HttpServletRequest request, HttpServletResponse response) {
        try {
            /* BEGIN-AUTOGENERATED-SERVLET-TESTS-9 */

            /* END-AUTOGENERATED-SERVLET-TESTS-9 */
        } catch (Throwable t) {

            /* this will only happen on an IOException or something (the runTest for the test cases will catch any exceptions from the test cases).  So, we just abort
             * and write to the console (since we can't write to the response without possibly throwing another exception)
             */

            System.out.println(
                    " Caught thowable from doGetCWE9 ");

            System.out.println(" Throwable's message = " + t.getMessage());

            System.out.println("Stack trace below");

            StackTraceElement stes[] = t.getStackTrace();

            for (StackTraceElement ste : stes) {
                System.out.println("  " + ste.toString());
            }
        }
    }
}

